app/frontend/src/features/auth/login/LoginForm.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  Box,
+  FormControlLabel,
+  InputLabel,
+  makeStyles,
+  Switch,
+  Typography,
+} from "@material-ui/core";
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+import { Link, useHistory } from "react-router-dom";
+
+import Button from "../../../components/Button";
+import TextBody from "../../../components/TextBody";
+import TextField from "../../../components/TextField";
+import { LoginRes } from "../../../pb/auth_pb";
+import { loginPasswordRoute, resetPasswordRoute } from "../../../routes";
+import { service } from "../../../service";
+import { useIsMounted, useSafeState } from "../../../utils/hooks";
+import { useAuthContext } from "../AuthProvider";
+import useAuthStyles from "../useAuthStyles";
+
+const useStyles = makeStyles((theme) => ({
+  loginOptions: {
+    display: "flex",
+    marginTop: theme.spacing(2),
+  },
+  forgotPasswordLink: {
+    color: "#333333",
+    lineHeight: theme.spacing(5),
+  },
+}));
+
+export default function UsernameForm() {
+  const classes = useStyles();
+  const authClasses = useAuthStyles();
+  const { authState, authActions } = useAuthContext();
+  const authLoading = authState.loading;
+  const [sent, setSent] = useState(false);
+  const [loading, setLoading] = useSafeState(useIsMounted(), false);
+  const [loginWithLink, setLoginWithLink] = useState(true);
+
+  const { handleSubmit, register } = useForm<{ username: string }>();
+  const history = useHistory();
+
+  const onSubmit = handleSubmit(
+    async (data: { username: string; password: string }) => {
+      setLoading(true);
+      authActions.clearError();
+      try {
+        const next = await service.auth.checkUsername(data.username);
+        switch (next) {
+          case LoginRes.LoginStep.INVALID_USER:
+            authActions.authError("Couldn't find that user.");
+            break;
+
+          case LoginRes.LoginStep.NEED_PASSWORD:
+            history.push(loginPasswordRoute, data);
+            //return here to avoid setLoading when
+            //this component is no longer mounted
+            setLoginWithLink(false);
+            return;
```

Comment:
```suggestion
            break;
```
Now that it's one page, you do need to setLoading(false) here or you can't login with a password. So no return.